File: DailyDataServer/src/main/java/com/pseandroid2/dailydataserver/postDatabase/Request/PostPreviewWrapper.java
/*

    DailyData is an android app to easily create diagrams from data one has collected
    Copyright (C) 2022  Antonia Heiming, Anton Kadelbach, Arne Kuchenbecker, Merlin Opp, Robin Amman

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.

*/
package com.pseandroid2.dailydataserver.postDatabase.Request;


import lombok.Getter;
import lombok.Setter;

import java.util.Arrays;


/**
 * This method contains everything, what is necessary fpr the post Preview.
 * It is used for a request, to get a post preview.
 */
@Getter
@Setter
public class PostPreviewWrapper {
    private Byte[] previewPicture;
    private String title;
    /**
     * The Constructor for the postPreviewWrapper.
     *
     * @param previewPicture the picture, which the user can see, as preview for the whole post.
     * @param title          the title of the post.
     */
    public PostPreviewWrapper(final Byte[] previewPicture, final String title) {
        this.previewPicture = previewPicture;
        this.title = title;
    }


    @Override
    public String toString() {
        return "PostPreviewWrapper{" +
                "PreviewPicture=" + Arrays.toString(previewPicture) +
                ", title='" + title + '\'' +
                '}';
    }
}

File: DailyData/app/src/main/java/com/pseandroid2/dailydata/remoteDataSource/serverConnection/serverParameter/PostPreviewWrapper.kt
/*

    DailyData is an android app to easily create diagrams from data one has collected
    Copyright (C) 2022  Antonia Heiming, Anton Kadelbach, Arne Kuchenbecker, Merlin Opp, Robin Amman

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.

*/
/*

    DailyData is an android app to easily create diagrams from data one has collected
    Copyright (C) 2022  Antonia Heiming, Anton Kadelbach, Arne Kuchenbecker, Merlin Opp, Robin Amman

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.

*/

package com.pseandroid2.dailydata.remoteDataSource.serverConnection.serverParameter

/**
 * A data class for uploading posts. It has the post preview picture and it's title
 *
 * @param previewPicture:   The preview picture of the post
 * @param title:            The title of the post
 */
data class PostPreviewWrapper(val previewPicture: ByteArray = ByteArray(0),
                              val title: String = "") {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as PostPreviewWrapper

        if (!previewPicture.contentEquals(other.previewPicture)) return false
        if (title != other.title) return false

        return true
    }

    override fun hashCode(): Int {
        var result = previewPicture.contentHashCode()
        result = 31 * result + title.hashCode()
        return result
    }
}
